src/utils/dockerContextManager.ts

```diff
@@ -0,0 +1,190 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExecOptions } from 'child_process';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+import * as url from 'url';
+import { workspace, WorkspaceConfiguration } from 'vscode';
+import { parseError } from "vscode-azureextensionui";
+import { ext } from '../extensionVariables';
+import { localize } from '../localize';
+import { LocalOSProvider } from './LocalOSProvider';
+import { execAsync } from './spawnAsync';
+import { timeUtils } from './timeUtils';
+
+const ContextInspectExecOptions: ExecOptions = { timeout: 5000 }
+const DOCKER_CONTEXT: string = 'DOCKER_CONTEXT';
+const DefaultRefreshIntervalSec: number = 20;
+const osp = new LocalOSProvider();
+const DockerConfigFilePath: string = osp.pathJoin(osp.os, osp.homedir, '.docker', 'config.json');
```

Comment:
Sounds good. The rationale for this change is that I needed to construct a file path in a platform-neutral way and the `LocalOSProvider` functionality was a very good fit. 

Also, `LocalOSProvider` really isn't specific to debugging in any way, and in this case I am just using it to initialize the manager, so no registration is taking place.

Reply:
Surprisingly it wasn't the registering, but all the constructing, that was slowing things down.

Reply:
Something fishy must be going on here. `LocalOSProvider` doesn't even have a constructor defined. And it has no state. It is just a bunch of methods.

Reply:
It wasn't only `LocalOSProvider`, there were a couple dozen parts being constructed and assembled.

Reply:
So @philliphoff what do you think about lifting `LocalOSProvider` into utilities?

Reply:
I think it's fine to lift `LocalOSProvider` out of the debugging folder and make it a more generalized service.  I'm not sure I'd switch to `os` directly (yet) until I saw the data that new'ing even a couple dozen objects was, itself, a significant performance hit (and not a consequence of something else being done in `registerDebugConfigurationProvider()`).  Even then, just dispensing with that individual service wouldn't (one would think) save any significant amount of time, so it would still have benefit from being consistent with the DI pattern of the other components.  (As opposed to a larger refactor to optimize load/init performance where we might rethink the DI pattern altogether.)

Reply:
@philliphoff @karolz-ms Sounds good. In any case, by _far_ the majority of the activation time is in loading/compiling the JS--by a couple orders of magnitude. The only ways we can improve on that are to reduce our dependencies and to lazy-load, if that's possible.